fe/fe-core/src/main/java/com/starrocks/planner/HudiScanNode.java

```diff
@@ -0,0 +1,419 @@
+        }
+        LOG.debug("partition prune cost: {} ms, partitions: {}",
+                (System.currentTimeMillis() - start), selectedPartitionIds);
+
+        noEvalPartitionConjuncts.addAll(((ListPartitionPruner) partitionPruner).getNoEvalConjuncts());
+    }
+
+    public void setSelectedPartitionIds(Collection<Long> selectedPartitionIds) {
+        this.selectedPartitionIds = selectedPartitionIds;
+    }
+
+    public void setIdToPartitionKey(Map<Long, PartitionKey> idToPartitionKey) {
+        this.idToPartitionKey = idToPartitionKey;
+    }
+
+    public List<Expr> getNonPartitionConjuncts() {
+        return nonPartitionConjuncts;
+    }
+
+    public List<Expr> getNoEvalPartitionConjuncts() {
+        return noEvalPartitionConjuncts;
+    }
+
+    public List<Expr> getMinMaxConjuncts() {
+        return minMaxConjuncts;
+    }
+
+    public void setMinMaxTuple(TupleDescriptor tuple) {
+        minMaxTuple = tuple;
+    }
+
+    public void getScanRangeLocations(DescriptorTable descTbl) throws UserException {
+        if (selectedPartitionIds.isEmpty()) {
+            return;
+        }
+
+        long start = System.currentTimeMillis();
+        List<PartitionKey> partitionKeys = Lists.newArrayList();
+        List<DescriptorTable.ReferencedPartitionInfo> partitionInfos = Lists.newArrayList();
+        for (long partitionId : selectedPartitionIds) {
+            PartitionKey partitionKey = idToPartitionKey.get(partitionId);
+            partitionKeys.add(partitionKey);
+            partitionInfos.add(new DescriptorTable.ReferencedPartitionInfo(partitionId, partitionKey));
+        }
+        List<String> partNames = Lists.newArrayListWithCapacity(partitionKeys.size());
+        List<String> partColumnNames = hudiTable.getPartitionColumnNames();
+        for (int i = 0; i < partitionKeys.size(); i++) {
+            descTbl.addReferencedPartitions(hudiTable, partitionInfos.get(i));
+            List<String> partValues = Utils.getPartitionValues(partitionKeys.get(i));
+            if (partValues.size() > 0) {
+                partNames.add(FileUtils.makePartName(partColumnNames, partValues));
+            } else {
+                partNames.add("");
+            }
+        }
+
+        Configuration conf = new Configuration();
+        HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder().setConf(conf).setBasePath(hudiTable.getHudiBasePath()).build();
+        HoodieEngineContext engineContext = new HoodieLocalEngineContext(conf);
+        HoodieMetadataConfig metadataConfig = HoodieMetadataConfig.newBuilder()
+                .enable(Boolean.parseBoolean(hudiTable.getHudiMatadataEnable()))
+                .build();
+        HoodieTableFileSystemView fileSystemView = FileSystemViewManager.createInMemoryFileSystemView(engineContext, metaClient, metadataConfig);
+        HdfsFileFormat hdfsFileFormat = HdfsFileFormat.fromHdfsInputFormatClass(hudiTable.getHudiTableInputFormat());
```

Comment:
We still use InputFormat for this version, right?

Reply:
just retrive file format from input format instead of using input format api here